plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -317,10 +332,17 @@ func (s *SQLServer) gatherServer(pool *sql.DB, query Query, acc telegraf.Accumul
 			return err
 		}
 	}
+
+	if rows.Err() == nil && query.HasCachedData && rows.NextResultSet() && rows.Next() {
+		var newCachedData string
+		rows.Scan(&newCachedData)
```

Comment:
need to check error

Reply:
Addressed the issue in new PR for SQL DB - #9150